drivers/crc/crc_hw.h

```diff
@@ -0,0 +1,253 @@
+	 * thread-safe mode
+	 */
+#ifdef CONFIG_CRC_THREADSAFE
+	struct crc_data *data = (struct crc_data *)dev->data;
+	int mtx = crc_lock(data->crc_8_timeout);
+
+	if (mtx != 0) {
+		return mtx;
+	}
+#endif /* CONFIG_CRC_THREADSAFE */
+
+	return api->crc_8_ccitt(dev, initial_value, buf, len);
+}
+
+/**
+ * @brief Verify 8-bit CRC CCITT of array *buf of length len
+ * @param expected Expected CRC result
+ * @param initial_value Initial value to be preloaded into CRC calculation unit
+ * @param *buf Pointer to array to be CRC'd
+ * @param len Number of bytes in *buf
+ * @retval Calculated 8-bit CRC CCITT
+ */
+static inline uint8_t verify_crc_8_ccitt(uint8_t expected, uint8_t initial_value, const void *buf,
+					 size_t len)
+{
+	static const struct device *dev = DEVICE_DT_GET(DT_NODELABEL(crc1));
+	const struct crc_driver_api *api = (const struct crc_driver_api *)dev->api;
+
+	return api->verify_crc_8_ccitt(dev, expected, initial_value, buf, len);
+}
+
+/**
+ * @brief Configure CRC unit to calculate CRC 32 IEEE
+ * @retval See crc_lock()
+ */
+static inline int setup_crc_32_ieee(void)
+{
+	static const struct device *dev = DEVICE_DT_GET(DT_NODELABEL(crc1));
+	const struct crc_driver_api *api = (const struct crc_driver_api *)dev->api;
+
+	/**
+	 * Attempt to take mutex lock, if we fail then return crc_lock() retvals,
+	 * else return the setup function which will return 0
+	 */
+#ifdef CONFIG_CRC_THREADSAFE
+	const struct crc_data *data = dev->data;
+	int mtx = crc_lock(data->crc_32_timeout);
+
+	if (mtx != 0) {
+		return mtx;
+	}
+#endif /* CONFIG_CRC_THREADSAFE */
+
+	return api->setup_crc_32_ieee(dev);
+}
+
+/**
+ * @brief Get 32-bit CRC on array *buf of length len
+ * @param *buf Pointer to array to be CRCd
+ * @param len Number of bytes in *buf
+ * @retval Calculated 32-bit CRC
+ */
+static inline uint32_t crc_32_ieee(const void *buf, size_t len)
+{
+	static const struct device *dev = DEVICE_DT_GET(DT_NODELABEL(crc1));
+	const struct crc_driver_api *api = (const struct crc_driver_api *)dev->api;
+
+	/**
+	 * We must check that the calling thread has access to the resource if in
+	 * thread-safe mode
+	 */
+#ifdef CONFIG_CRC_THREADSAFE
+	struct crc_data *data = (struct crc_data *)dev->data;
+	int mtx = crc_lock(data->crc_32_timeout);
+
+	if (mtx != 0) {
+		return mtx;
+	}
+#endif /* CONFIG_CRC_THREADSAFE */
+
+	return api->crc_32_ieee(dev, buf, len);
+}
+
+/**
+ * @brief Verify 32-bit CRC of array *buf of length len
+ * @param expected Expected CRC result
+ * @param *buf Pointer to array to be CRC'd
+ * @param len Number of bytes in *buf
+ * @retval Calculated 32-bit CRC
+ */
+static inline uint32_t verify_crc_32_ieee(uint32_t expected, const void *buf, size_t len)
+{
+	static const struct device *dev = DEVICE_DT_GET(DT_NODELABEL(crc1));
+	const struct crc_driver_api *api = (const struct crc_driver_api *)dev->api;
+
+	return api->verify_crc_32_ieee(dev, expected, buf, len);
+}
+
+#endif /* CONFIG_CRC_HW */
+
```

Comment:
The Doxygen group ending comment is missing
```suggestion
/**
 * @}
 */
```